code/go/0chain.net/chaincore/block/entity.go

```diff
@@ -179,6 +180,13 @@ func NewBlock(chainID datastore.Key, round int64) *Block {
 	return b
 }
 
+func (b *Block) GetUniqueBlockExtensions() (uBlExts map[string]bool) {
+	b.uniqueBlockExtMutex.RLock()
+	defer b.uniqueBlockExtMutex.RUnlock()
+
+	return b.uniqueBlockExtensions
```

Comment:
We need to return the copy of the map, this returning would simply expose the map to public without locks protection.